R/score_transition_risk_and_polish.R

```diff
@@ -0,0 +1,203 @@
+        "matched_reference_product",
+        "unit",
+        "co2e_lower",
+        "co2e_upper",
+        "emission_profile",
+        "benchmark",
+        "profile_ranking",
+        "tilt_sector",
+        "tilt_subsector",
+        "co2_footprint"
+      )
+    )
+  select_sector_profile_at_product_level <- sector_profile_at_product_level |>
+    select(
+      c(
+        "companies_id",
+        "ep_product",
+        "sector_profile",
+        "scenario",
+        "year",
+        "reduction_targets"
+      )
+    )
+
+  select_transition_risk_score_at_product_level <- transition_risk_score_at_product_level |>
+    select(c(
+      "companies_id",
+      "ep_product",
+      "benchmark_tr_score",
+      "transition_risk_score"
+    ))
+
+  bundesbank_data_at_product_level <- select_emissions_profile_at_product_level |>
+    left_join(
+      select_sector_profile_at_product_level,
+      relationship = "many-to-many",
+      by = c("companies_id", "ep_product")
+    ) |>
+    mutate(benchmark_tr_score = paste(scenario, year, benchmark, sep = "_")) |>
+    left_join(select_transition_risk_score_at_product_level, by = c("companies_id", "ep_product", "benchmark_tr_score")) |>
+    distinct()
+    ## TODO: `europages_companies` should include headcount? Submit issue to tiltIndicatorBefore
+    # left_join(headcount, by = c("companies_id")) |>
+    ## This is specific to the latest bundesbank delivery
+    # filter(country == "germany", benchmark %in% c("unit", "unit_tilt_sector") | is.na(benchmark))
+
+  select_emissions_profile_at_company_level <- emissions_profile_at_company_level |>
+    select(
+      c(
+        "companies_id",
+        "country",
+        "main_activity",
+        "co2e_lower",
+        "co2e_upper",
+        "benchmark",
+        "emission_profile",
+        "emission_profile_share",
+        "profile_ranking_avg",
+        "co2_avg"
+      )
+    )
+
+  select_sector_profile_at_company_level <- sector_profile_at_company_level |>
+    select(c("companies_id", "scenario", "year", "reduction_targets_avg"))
+
+  select_transition_risk_score_at_company_level <- transition_risk_score_at_company_level |>
+    select(
+      c(
+        "companies_id",
+        # `benchmark_tr_score` was renamed to `benchmark_tr_score_avg`
+        # https://2degreesinvesting.github.io/tiltIndicatorAfter/news/index.html#tiltindicatorafter-0009026-2024-04-04
+        matches("benchmark_tr_score"),
+        "transition_risk_score_avg"
+      )
+    )
+
+  tmp <- select_emissions_profile_at_company_level
+  if (pivot_wider) {
+    tmp <- tmp |>
+      exclude_cols_then_pivot_wider(
+        exclude_cols = "co2e",
+        id_cols = c(
+          companies_id,
+          country,
+          main_activity,
+          benchmark,
+          profile_ranking_avg,
+          co2_avg
+        ),
+        names_from = emission_profile,
+        names_prefix = "emission_category_",
+        values_from = emission_profile_share
+      )
+  }
```

Comment:
@maurolepore After applying `pivot_wider()` you didn't use `unchop(everything()) |>` command as I used in the script. I am not sure whether you missed it or excluded in by purpose. 

Nevertheless, I used it to convert all the list type columns into normal string type columns after `pivot_wider` create these list columns. All the four "emissions_category_" columns will become list columns by default, and I couldn't find a better solution to make them normal columns other than using the `unchop` function. So, please investigate and remove this issue maybe with a better solution. Thanks!

Reply:
FYI https://github.com/2DegreesInvesting/tiltIndicatorAfter/pull/206.